source/containers.tex

```diff
@@ -25572,62 +25714,166 @@
 \begin{itemize}
 \item
-\tcode{$s_k$.extent == 0 ? 0 : 1 + (\exposid{de-ice}($s_k$.extent) - 1) / \exposid{de-ice}($s_k$.stride)}
-if $S_k$ is a specialization of \tcode{strided_slice},
-
+  \tcode{$\sigma_k$.extent == 0 ? 0 : 1 + ($\sigma_k$.extent - 1) / $\sigma_k$.stride}
+  if the type of $\sigma_k$ is a specialization of \tcode{strided_slice},
 \item
-otherwise,
-\tcode{\exposid{last_}<$k$>(src, slices...) - \exposid{first_}<IndexType, $k$>(slices...)}.
+  otherwise,
+  $U - L$, where \range{$L$}{$U$} is the \tcode{submdspan} slice range
+  of $\sigma_k$ for the $k^\text{th}$ extent of \tcode{src}.
 \end{itemize}
 \end{itemdescr}
 
 \rSec4[mdspan.sub.map]{Specializations of \tcode{submdspan_mapping}}
 
-\rSec5[mdspan.sub.map.common]{Common}
+\rSec5[mdspan.sub.map.sliceable]{Sliceable layout mapping requirements}
 
 \pnum
-The following elements apply to all functions in \ref{mdspan.sub.map}.
+Let:
+\begin{itemize}
+\item
+  \tcode{M} denote a layout mapping class;
+\item
+  \tcode{IT} denote \tcode{M::extent_type::index_type};
+\item
+  \tcode{m} denote a value of type (possibly const) \tcode{M};
+\item
+  \tcode{M_rank} be equal to \tcode{M::extent_type::rank()};
+\item
+  \tcode{valid_slices} denote a pack of (possibly const) objects
+  for which \tcode{sizeof...(valid_slices) == M_rank} is \tcode{true} and,
+  for each rank index $i$ of \tcode{m.extents()},
+  \tcode{valid_slices...[i]} is a valid \tcode{submdspan} slice
+  for the $i^\text{th}$ extent of \tcode{m.extents()};
+\item
+  \tcode{invalid_slices} denote a pack of objects
+  for which \tcode{sizeof...(invalid_slices) == M_rank} is \tcode{true} and
+  there exists an integer $k$ such that the cv-unqualified type
+  of \tcode{invalid_slices...[$k$]} is none of the following:
+  \begin{itemize}
+  \item \tcode{IT},
+  \item \tcode{full_extent_t},
+  \item a specialization of \tcode{constant_wrapper}, or
+  \item a specialization of \tcode{strided_slice}.
+  \end{itemize}
+\end{itemize}
 
 \pnum
-\constraints
-\tcode{sizeof...(slices)} equals \tcode{extents_type::rank()}.
+For the purpose of this section,
+the meaning of \tcode{submdspan_mapping} is established
+as if by performing argument-dependent lookup only\iref{basic.lookup.argdep}.
 
 \pnum
-\mandates
-For each rank index $k$ of \tcode{extents()},
-exactly one of the following is true:
+A type \tcode{M} meets the \defn{sliceable layout mapping requirements} if
 \begin{itemize}
 \item
-$S_k$ models \tcode{\libconcept{convertible_to}<index_type>},
+\tcode{M} meets the layout mapping requirements\iref{mdspan.layout.policy.reqmts},
+\item
+the expression \tcode{submdspan_mapping(m, invalid_slices...)} is ill-formed, and
 \item
-$S_k$ models \tcode{\exposconcept{index-pair-like}<index_type>},
+%FIXME: is ISO going to like a bullet ending in a colon?
+the following expression is well-formed and has the specified semantics:
+\end{itemize}
+
+\begin{itemdecl}
+submdspan_mapping(m, valid_slices...)
+\end{itemdecl}
```

Comment:
This is an expression being shown for the preceding sentence "the following expression is well-formed and has the specified semantics:"

It's not the start of a new \itemdecl

I think it should be a codeblock environment instead ... although then it's a bit weird having the _Result_ and _Returns_ elements for an expression.